library/src/main/java/org/apache/apex/malhar/lib/fs/s3/S3RecordReader.java

```diff
@@ -0,0 +1,604 @@
+    delimitedRecordReaderContext.getS3Params().initializeS3Client(accessKey, secretAccessKey, endPoint);
+    delimitedRecordReaderContext.getS3Params().setBucketName(bucketName);
+    delimitedRecordReaderContext.setOverflowBufferSize(overflowBufferSize);
+    return delimitedRecordReaderContext;
+  }
+
+  /**
+   * S3RecordReaderParams is used to hold the common parameters used by the
+   * DelimitedRecordReaderContext and FixedWidthReacordReaderContext for S3
+   */
+  protected static class S3RecordReaderParams
+  {
+    /**
+     * Amazon client used to read bytes from S3
+     */
+    private AmazonS3 s3Client;
+    /**
+     * S3 bucket name
+     */
+    private String bucketName;
+    /**
+     * path of file being processed in bucket
+     */
+    private String filePath;
+    /**
+     * length of the file being processed
+     */
+    private long fileLength;
+
+    /**
+     * Initialize the AmazonS3 client using the accessKey, secretAccessKey, sets
+     * endpoint for the s3Client if provided
+     *
+     * @param accessKey
+     * @param secretAccessKey
+     * @param endPoint
+     */
+    public void initializeS3Client(String accessKey, String secretAccessKey, String endPoint)
+    {
+      s3Client = new AmazonS3Client(new BasicAWSCredentials(accessKey, secretAccessKey));
+      if (endPoint != null) {
+        s3Client.setEndpoint(endPoint);
+      }
+    }
+
+    /**
+     * Set the AmazonS3 service
+     *
+     * @param s3Client
+     *          given s3Client
+     */
+    public void setS3Client(@javax.validation.constraints.NotNull AmazonS3 s3Client)
+    {
+      Preconditions.checkNotNull(s3Client);
+      this.s3Client = s3Client;
+    }
+
+    /**
+     * Returns the AmazonS3 service
+     *
+     * @return s3Client
+     */
+    public AmazonS3 getS3Client()
+    {
+      return s3Client;
+    }
+
+    /**
+     * Set the bucket name
+     *
+     * @param bucketName
+     *          given bucketName
+     */
+    public void setBucketName(@javax.validation.constraints.NotNull String bucketName)
+    {
+      Preconditions.checkNotNull(bucketName);
+      this.bucketName = bucketName;
+    }
+
+    /**
+     * Returns the bucket name
+     *
+     * @return bucketName
+     */
+    public String getBucketName()
+    {
+      return bucketName;
+    }
+
+    /**
+     * Sets the file path
+     *
+     * @param filePath
+     *          given filePath
+     */
+    public void setFilePath(String filePath)
+    {
+      this.filePath = filePath;
+    }
+
+    /**
+     * Returns the file path
+     *
+     * @return filePath
+     */
+    public String getFilePath()
+    {
+      return filePath;
+    }
+
+    /**
+     * Sets the length of the file to which the block belongs
+     *
+     * @param fileLength
+     *          length of the file to which the block belongs
+     */
+    public void setFileLength(long fileLength)
+    {
```

Comment:
I think there is no use of setFilePath(), setFileLength(). Can you remove the setter methods for filePath and fileLength ?